server/controllers/userController.js

```diff
@@ -0,0 +1,33 @@
+const jwt = require("jsonwebtoken");
+const User = require("../models/User");
+
+exports.create = async (req, res, next) => {
+  try {
+    const { username, password, email } = req.body;
+    const user = new User({
+      username,
+      email,
+      password,
+    });
+    const response = await user.save();
+    const token = jwt.sign({ user }, process.env.JWT_SECRET);
+    return res.status(201).cookie("token", token, { httpOnly: true }).json({
```

Comment:
You'd want to make this a secure cookie for production environment (can use env variables to tell)